lib/filesystem/FileStream.cpp

```diff
@@ -0,0 +1,148 @@
+#include "StdInc.h"
+#include "FileStream.h"
+#include "../minizip/unzip.h"
+
+#include <cstdio>
+
+
+#ifdef _WIN32
+	#ifndef _CRT_SECURE_NO_WARNINGS
+		#define _CRT_SECURE_NO_WARNINGS
+	#endif
+	#include <cwchar>
+	#define CHAR_LITERAL(s) L##s
+	#define DO_OPEN(name, mode) _wfopen(name, mode)
+	using CharType = wchar_t;
+#else
+	#define CHAR_LITERAL(s) s
+	#define DO_OPEN(name, mode) fopen(name, mode)
+	using CharType = char;
+#endif
+
+#define GETFILE static_cast<std::FILE*>(filePtr)
+
+voidpf ZCALLBACK MinizipOpenFunc(voidpf opaque, const void* filename, int mode)
+{
+    const CharType* mode_fopen = [mode]() -> const CharType*
+    {
+		if ((mode & ZLIB_FILEFUNC_MODE_READWRITEFILTER) == ZLIB_FILEFUNC_MODE_READ)
+			return CHAR_LITERAL("rb");
+		else if (mode & ZLIB_FILEFUNC_MODE_EXISTING)
+			return CHAR_LITERAL("r+b");
+		else if (mode & ZLIB_FILEFUNC_MODE_CREATE)
+			return CHAR_LITERAL("wb");
+		return nullptr;
+    }();
+
+    if (filename != nullptr && mode_fopen != nullptr)
```

Comment:
I assign it nullptr if mode (the integer) has an invalid value. This is similar to how minizip does it internally; if an invalid mode is used, the FILE* will become a nullptr.